Jellyfin.Server.Implementations/Users/UserManager.cs

```diff
@@ -0,0 +1,818 @@
+            ILogger<IUserManager> logger)
+        {
+            _dbProvider = dbProvider;
+            _cryptoProvider = cryptoProvider;
+            _networkManager = networkManager;
+            _appHost = appHost;
+            _imageProcessor = imageProcessor;
+            _logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserPasswordChanged;
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserUpdated;
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserCreated;
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserDeleted;
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserLockedOut;
+
+        /// <inheritdoc/>
+        public IEnumerable<User> Users
+        {
+            get
+            {
+                var dbContext = _dbProvider.CreateContext();
+                return dbContext.Users;
+            }
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<Guid> UsersIds
+        {
+            get
+            {
+                var dbContext = _dbProvider.CreateContext();
+                return dbContext.Users.Select(u => u.Id);
+            }
+        }
+
+        /// <inheritdoc/>
+        public User GetUserById(Guid id)
+        {
+            if (id == Guid.Empty)
+            {
+                throw new ArgumentException("Guid can't be empty", nameof(id));
+            }
+
+            var dbContext = _dbProvider.CreateContext();
+
+            return dbContext.Users.Find(id);
+        }
+
+        /// <inheritdoc/>
+        public User GetUserByName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException("Invalid username", nameof(name));
+            }
+
+            var dbContext = _dbProvider.CreateContext();
+
+            // This can't use an overload with StringComparer because that would cause the query to
+            // have to be evaluated client-side.
+            return dbContext.Users.FirstOrDefault(u => string.Equals(u.Username, name));
+        }
+
+        /// <inheritdoc/>
+        public async Task RenameUser(User user, string newName)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            if (string.IsNullOrWhiteSpace(newName))
+            {
+                throw new ArgumentException("Invalid username", nameof(newName));
+            }
+
+            if (user.Username.Equals(newName, StringComparison.Ordinal))
+            {
+                throw new ArgumentException("The new and old names must be different.");
+            }
+
+            if (Users.Any(u => u.Id != user.Id && u.Username.Equals(newName, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Should this be running client-side?

Reply:
I directly ported the old behavior, if we want to change it, it should be in another pr imo